### PR TITLE
Ctctraders 1098 journeymodel

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/models/journeyDomain/RouteDetails.scala [hunk 1]
```diff
@@ -39,26 +40,41 @@ object RouteDetails {
...
   object TransitInformation {
 
-    implicit val reads: Reads[TransitInformation] =
-      ((__ \ AddAnotherTransitOfficePage.key).read[String] and
-        (__ \ ArrivalTimesAtOfficePage.key \ "dateTime").read[LocalDateTime])(TransitInformation(_, _))
+    implicit val readSeqTransitInformation: UserAnswersReader[NonEmptyList[TransitInformation]] =
+      AddSecurityDetailsPage.reader
+        .flatMap {
+          addSecurityDetailsFlag =>
+            if (addSecurityDetailsFlag) {
```
Comment (chris-ianson) [inline app/models/journeyDomain/RouteDetails.scala:52]: Could the if condition come within the case on line 56 so you don't have two blocks of code which are doing almost the same thing (only difference being if we're setting a date on TransitInformation)

File: app/models/journeyDomain/TransportDetails.scala [hunk 2]
```diff
@@ -127,11 +156,36 @@ object TransportDetails {
...
+    implicit val reader: UserAnswersReader[ModeCrossingBorder] =
+      ModeCrossingBorderPage.reader.flatMap(
+        modeCode =>
+          if ((Mode5or7.Constants.codes ++ Rail.Constants.codes).contains(modeCode))
+            ModeExemptNationality.pure[UserAnswersReader].widen[ModeCrossingBorder]
+          else
+            NationalityCrossingBorderPage.reader.map(ModeWithNationality(_))
+      )
+
     object ModeExemptNationality extends ModeCrossingBorder // 2, 20, 5, 50, 7, 70
```
Comment (chris-ianson) [inline app/models/journeyDomain/TransportDetails.scala:189]: Do we need this comment?

File: app/models/messages/header/Transport.scala [hunk 3]
```diff
@@ -58,7 +58,7 @@ object Transport {
 
   implicit val reads: XmlReader[Transport] = (
     (__ \\ "InlTraModHEA75").read[Int].optional,
-    (__ \\ "TraModAtBorHEA76").read[Int].optional,
+    (__ \\ "TraModAtBorHEA76").read[String].optional,
```
Comment (chris-ianson) [inline app/models/messages/header/Transport.scala:61]: Why is this String? If it stays as string we can remove toString on line 47

**Review by chris-ianson — APPROVED** [review]
Some minor comments. I know DeclarationRequestService still needs work to get the other data in so haven't commented on static values/comments.

File: app/models/journeyDomain/TransportDetails.scala (hunk 2 above)
Comment (amaalali) [reply app/models/journeyDomain/TransportDetails.scala:189]: Not any more, this is now in a constant in this file, so it's outdated

File: app/models/messages/header/Transport.scala (hunk 3 above)
Comment (amaalali) [reply app/models/messages/header/Transport.scala:61]: We will need to do this 👍 
@MatthewAlexanderBeck reminder for us to address in the next PR

**Pull request merged**